lib/collection/src/shard/replica_set.rs

```diff
@@ -17,27 +22,28 @@ pub type IsActive = bool;
 
 /// A set of shard replicas.
 /// Handles operations so that the state is consistent across all the replicas of the shard.
+/// Prefers local shard for read-only operations.
+/// Perform updates on all replicas and report error if there is at least one failure.
 pub struct ReplicaSet {
     shard_id: ShardId,
     this_peer_id: PeerId,
     local: Option<LocalShard>,
-    // TODO: Remote shard should be able to query several peers
-    remote: Option<RemoteShard>,
-    pub(crate) replica_state: HashMap<PeerId, IsActive>,
+    remotes: Vec<RemoteShard>,
+    pub(crate) replica_state: RwLock<HashMap<PeerId, IsActive>>,
```

Comment:
Let's also add `SaveOnDisk<HashMap<PeerId, IsActive>>` as we discussed we need to retain these info. Of course this can also be done in a separate PR

Reply:
Good point, I will do this in a follow up PR to keep things small :+1: